### PR TITLE
feat: add new step tracking approach allowing for custom steps

### PR DESCRIPTION
**Pull request by matthewgallo**
Contributes to #1327 #1364 #1511 #1512 

This PR changes how the **CreateTearsheet** and **CreateFullPage** components keep track of steps, allowing us to more easily allow users to utilize custom steps. These components are both using React's context api which allows data from the step component to be accessed from the parent (and we can know and trust that it's coming from the step component).

Because of the fact that both the **CreateTearsheet** and the **CreateFullPage** use the same custom hooks, I had to make this change to both of these components at the same time. I also removed the old code that wasn't being used and was never released specific to the CreateTearsheetSection/CreateFullPageSection components. This allowed me to clean up the code for these components a lot.

#### What did you change?
```
packages/cloud-cognitive/src/components/CreateFullPage/CreateFullPage.js
packages/cloud-cognitive/src/components/CreateFullPage/CreateFullPage.test.js
packages/cloud-cognitive/src/components/CreateFullPage/CreateFullPageSection.js
packages/cloud-cognitive/src/components/CreateFullPage/CreateFullPageStep.js
packages/cloud-cognitive/src/components/CreateFullPage/_create-full-page.scss
packages/cloud-cognitive/src/components/CreateFullPage/constants.js
packages/cloud-cognitive/src/components/CreateFullPage/index.js

packages/cloud-cognitive/src/components/CreateInfluencer/CreateInfluencer.js
packages/cloud-cognitive/src/components/CreateInfluencer/CreateInfluencer.test.js

packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.js
packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.stories.js
packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.test.js
packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheetSection.js
packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheetStep.js
packages/cloud-cognitive/src/components/CreateTearsheet/_create-tearsheet.scss
packages/cloud-cognitive/src/components/CreateTearsheet/constants.js
packages/cloud-cognitive/src/components/CreateTearsheet/index.js
packages/cloud-cognitive/src/components/CreateTearsheet/preview-components/MultiStepTearsheet.js
packages/cloud-cognitive/src/components/CreateTearsheet/preview-components/MultiStepWithSectionsTearsheet.js

packages/cloud-cognitive/src/global/js/hooks/useCreateComponentFocus.js
packages/cloud-cognitive/src/global/js/hooks/useCreateComponentStepChange.js
packages/cloud-cognitive/src/global/js/hooks/useValidCreateStepCount.js
packages/cloud-cognitive/src/global/js/utils/hasValidType.js
```
#### How did you test and verify your work?
Storybook and ran tests for CreateTearsheet, CreateFullPage, and CreateInfluencer.

Shoutout to @dcwarwick for the [prototype](https://codesandbox.io/s/react-step-controller-using-contexts-9drf6)!

### CODE REVIEW

**Comment by davidmenendez** [comment]
clearly there's a lot going on here. just wanted to let you know i'm on it, but that it'll probably take me some time to sift through everything 👍

**Comment by matthewgallo** [comment]
Thanks @davidmenendez for reviewing. Indeed it's a lot, because CreateTearsheet/CreateFullPage share custom hooks, they both had to be updated together :/  Happy to answer any questions along the way

File: packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.js [hunk 1]
```diff
@@ -62,40 +60,40 @@ export let CreateTearsheet = forwardRef(
...
       previousState,
       currentStep,
-      getTearsheetSteps,
-      blockClass
-    );
-    useValidCreateStepCount(getTearsheetSteps, componentName);
+      blockClass,
+      onMount,
+    });
+    useValidCreateStepCount(stepData.length, componentName);
```
[the remark this answers is not in the document]
Comment (ninjasun) [reply packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.js:78]: Can the number of steps be dynamic?

[the remark this answers is not in the document]
Comment (matthewgallo) [reply packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.js:78]: Hey @ninjasun, we haven't considered dynamic steps at this point in time. But with this new approach, it should be something we can take a look at. If this is something your team is requiring, can you open a separate issue and provide an example/use case?

**Review by davidmenendez — APPROVED** [review]
i don't see anything glaringly wrong here, but there is a lot going on. definitely keep an eye out on this one. in general i'm not a fan of letting components get too big and out of control. hopefully this helps alleviate some of that.

**Comment by matthewgallo** [comment]
Yes, this definitely helped with reducing the size of some of these components. I'll be keeping a close eye on things 👀

**Pull request merged**